src/MTopJetPreSelectionModule3Jets.cxx

```diff
@@ -0,0 +1,204 @@
+
+  // bools
+  bool isMC;
+  bool debug;
+  
+};
+
+/*
+███    ███  ██████  ██████  ██    ██ ██      ███████
+████  ████ ██    ██ ██   ██ ██    ██ ██      ██
+██ ████ ██ ██    ██ ██   ██ ██    ██ ██      █████
+██  ██  ██ ██    ██ ██   ██ ██    ██ ██      ██
+██      ██  ██████  ██████   ██████  ███████ ███████
+*/
+
+MTopJetPreSelectionModule3Jets::MTopJetPreSelectionModule3Jets(uhh2::Context& ctx){
+
+  debug = string2bool(ctx.get("Debug","false")); // look for Debug, expect false if not found
+
+  //// CONFIGURATION
+  bool isherwig;
+  if(ctx.get("dataset_version") == "TTbar_powheg-herwig") isherwig = true;
+  else isherwig = false;
+
+  isMC = (ctx.get("dataset_type") == "MC"); 
+
+  if(debug) cout << "CONFIGURATION END" << endl;
+
+  //// HANDLES
+  h_recsel = ctx.declare_event_output<bool>("passed_recsel");
+  h_gensel = ctx.declare_event_output<bool>("passed_gensel");
+
+  h_fatjets = ctx.get_handle<std::vector<TopJet>>("xconeCHS");
+
+  if(debug) cout << "Output and Handles End" << endl;
+
+  //// COMMON MODULES
+
+  if(!isMC) lumi_sel.reset(new LumiSelection(ctx));
+
+  if(debug) cout << "LumiSelection End" << endl;
+
+  /* GEN M-ttbar selection [TTbar MC "0.<M^{gen}_{ttbar}(GeV)<700.] */
+  const std::string ttbar_gen_label("ttbargen");
+
+  ttgenprod.reset(new TTbarGenProducer(ctx, ttbar_gen_label, false));
+
+  if(debug) cout << "TTbarGenProducer" << endl;
+
+  TString samplename = ctx.get("dataset_version");
+  if( samplename.Contains("Mtt0000to0700") ){
+    genmttbar_sel.reset(new MttbarGenSelection(0., 700.));
+  }
+  else{
+    genmttbar_sel.reset(new uhh2::AndSelection(ctx));
+  }
+
+  //// EVENT SELECTION REC
+  met_sel.reset(new METCut(40, uhh2::infinity));
+  muon_sel.reset(new NMuonSelection(1, -1, MuonId(PtEtaCut(50, 2.4 ))));
+  elec_sel.reset(new NElectronSelection(1, -1, ElectronId(PtEtaCut(50, 2.4))));
+
+  //// EVENTS SELECTION GEN
+  if(isMC && !isherwig) SemiLepDecay.reset(new TTbarSemilep(ctx));
+  else if(isherwig) SemiLepDecay.reset(new TTbarSemilep_herwig(ctx));
+  if(isMC){
+    GenMuonPT.reset(new GenMuonSel(ctx, 55.));
+    GenElecPT.reset(new GenElecSel(ctx, 55.));
+  }
+
+}
+
+/*
+██████  ██████   ██████   ██████ ███████ ███████ ███████
+██   ██ ██   ██ ██    ██ ██      ██      ██      ██
+██████  ██████  ██    ██ ██      █████   ███████ ███████
+██      ██   ██ ██    ██ ██      ██           ██      ██
+██      ██   ██  ██████   ██████ ███████ ███████ ███████
+*/
+
+bool MTopJetPreSelectionModule3Jets::process(uhh2::Event& event){
+
+  if(debug) cout << "Start Module - Process" << endl;
+
+  /* CMS-certified luminosity sections */
+  if(event.isRealData){
+    if(!lumi_sel->passes(event)) return false;
+  }
+
+  if(!event.isRealData){
+    /* GEN M-ttbar selection */
+    ttgenprod->process(event);
+    if(!genmttbar_sel->passes(event)) return false;
+  }
+
+  //// EVENT SELECTION REC
+  bool passed_recsel;
+
+  bool pass_lep1 = ((event.muons->size() >= 1) || (event.electrons->size() >= 1));
```

Comment:
Use object names, that clearly state the purpose. That way one does not have to jump between lines.
```suggestion
  bool pass_lep_number = ((event.muons->size() >= 1) || (event.electrons->size() >= 1));
```